fuel-core-interfaces/src/txpool.rs

```diff
@@ -26,61 +26,66 @@ pub trait TxPoolDb:
     /// from p2p **RespondTransactions** or from userland. Because of userland we are returning
     /// error for every insert for better user experience.
-    async fn insert(&self, tx: Vec<Arc<Transaction>>)
-        -> Vec<anyhow::Result<Vec<Arc<Transaction>>>>;
-
+    Insert {
+        txs: Vec<Arc<Transaction>>,
+        response: oneshot::Sender<Vec<anyhow::Result<Vec<Arc<Transaction>>>>>,
+    },
     /// find all tx by their hash
-    async fn find(&self, hashes: &[TxId]) -> Vec<Option<TxInfo>>;
-
+    Find {
+        ids: Vec<TxId>,
+        response: oneshot::Sender<Vec<Option<TxInfo>>>,
+    },
     /// find one tx by its hash
-    async fn find_one(&self, hash: &TxId) -> Option<TxInfo>;
-
+    FindOne {
+        id: TxId,
+        response: oneshot::Sender<Option<TxInfo>>,
+    },
     /// find all dependent tx and return them with requsted dependencies in one list sorted by Price.
-    async fn find_dependent(&self, hashes: &[TxId]) -> Vec<Arc<Transaction>>;
-
+    FindDependent {
+        ids: Vec<TxId>,
+        response: oneshot::Sender<Vec<Arc<Transaction>>>,
+    },
+    /// remove transaction from pool needed on user demand. Low priority
+    Remove { ids: Vec<TxId> },
     /// Iterete over `hashes` and return all hashes that we dont have.
     /// Needed when we receive list of new hashed from peer with
     /// **BroadcastTransactionHashes**, so txpool needs to return
     /// tx that we dont have, and request them from that particular peer.
-    async fn filter_by_negative(&self, hashes: &[TxId]) -> Vec<TxId>;
-
-    /// Return all sorted transactions that are includable in next block.
-    /// This is going to be heavy operation, use it with only when needed.
-    async fn includable(&self) -> Vec<Arc<Transaction>>;
-
-    /// When block is updated we need to receive all spend outputs and remove them from txpool
-    /// There is three posibilities here. New Block is added or block is reverted/slashed from
-    ///  chain or both. When new block is added we need to remove transactions from pool.
-    /// Un case of block reverting we need to reinsert transaction in pool. What usually happens
-    /// with reorg is that one block get reverted and one new gets add, in that sense it is highly
-    /// probably that a lof of transactions that got reverted are going to be reinserted in new block.
-    /// That explains why we have only one function and it contains two list of reverted/inserted
-    /// transactions
-    async fn block_update(&self /*spend_outputs: [Input], added_outputs: [AddedOutputs]*/);
+    FilterByNegative {
+        ids: Vec<TxId>,
+        response: oneshot::Sender<Vec<TxId>>,
+    },
+    /// stop txpool
+    Stop,
+}
 
-    /// remove transaction from pool needed on user demand. Low priority
-    async fn remove(&self, hashes: &[TxId]);
+#[derive(Clone, Debug)]
+pub enum TxStatus {
+    /// Included into txpool.
+    Included,
```

Comment:
Included could be confusing terminology as Included is often used to say a transaction has been Included in a Block. You included a doc comment though so should be fine, just wanted to mention as it was my initial thought